src/realm/query_engine.cpp

```diff
@@ -684,3 +686,61 @@ ExpressionNode::ExpressionNode(const ExpressionNode& from, QueryNodeHandoverPatc
 , m_expression(from.m_expression->clone(patches))
 {
 }
+
+ReadAccessNode::ReadAccessNode(size_t acl_column_index, StringData user_id)
+    : m_acl_column(acl_column_index)
+    , m_user_id(user_id)
+{
+    m_dD = 10.0;
+    m_dT = 50.0;
+}
+
+void ReadAccessNode::table_changed()
+{
+    REALM_ASSERT(m_table->get_column_type(m_acl_column) == type_LinkList);
+    m_column = &m_table->get_column_link_list(m_acl_column);
+
+    m_permissions_table = const_cast<Table*>(m_table->get_link_target(m_acl_column).get());
+    m_read_col_ndx = m_permissions_table->get_column_index("canRead");
```

Comment:
I think according to sync, it is a normal and valid request to try to query with permissions when no permissions have been set up and in this case, assume that the user can read. Given that, we should either not assert here if the corresponding permissions columns can not be found, or else consider accepting the appropriate column indexes as inputs to this query node, since to verify that the permissions are there, the callers will have had to find them already.

Reply:
I will take this up with @simonask